app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {}, metadata = nil, analysis_id = nil) # rubocop:disable Metrics/ParameterLists
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = metadata
+        @analysis_id = analysis_id
+      end
+
+      def execute # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
+        authorize! sample.project, to: :update_sample?
+
+        if @project.id != @sample.project.id
+          raise SampleMetadataUpdateError,
+                I18n.t('services.samples.metadata.sample_does_not_belong_to_project', sample_name: @sample.name,
+                                                                                      project_name: @project.name)
+        end
+
+        if @metadata
+          @metadata = @metadata.transform_keys(&:to_s)
+          @metadata.each do |k, v|
+            if v.blank?
+              @sample['metadata'].delete(k) && @sample['metadata_provenance'].delete(k) if @sample['metadata'].key?(k)
+            else
+              @sample['metadata'][k] = v
```

Comment:
We probably should update the code here to not allow a user to be able to overwrite metadata fields/values that were previously added by an analysis. Another analysis could update it, but if it's a user update then we should probably not allow that. Thoughts @ericenns

Reply:
Yeah I agree a separate method to filter out the keys where the `metadata_provenance` indicates that it was last touched by an `analysis` should not overwritten by a user. Though a `User` should be able to remove those keys.

Reply:
Updated in e64d71ce473cbb8064cd28816d776d62536d2d07 and c789fc25fe68ba49a34ea3f76540e6597d3eed2d